test-dts/mount.d-test.ts

```diff
@@ -211,6 +218,78 @@ class ClassComponent extends Vue {
 expectError(mount(ClassComponent, {}).vm.changeMessage())
 mount(ClassComponent, {}).vm.changeMessage('')
 
+// region standard class component implement
+class StandardClassComponent<Props extends {} = {}> {
+  static defaultProps?: Record<string, Prop<any>> | string[]
+  private static __vccValue?: ComponentOptions
+  static get __vccOpts(): ComponentOptions {
+    if (this.__vccValue) return this.__vccValue
+    const CompConstructor = this
+    return (this.__vccValue = {
+      name: CompConstructor.name,
+      props: CompConstructor.defaultProps,
+      setup(props, ctx) {
+        const instance = new CompConstructor()
+        return instance.render.bind(instance)
+      }
+    })
+  }
+  constructor() {
+    const instance = getCurrentInstance()!
+    this.props = instance.props as Props
+    // @ts-expect-error no explicit setupContext on instance
+    this.context = instance.setupContext as SetupContext
+  }
+
+  props: Props
+  get $props() {
+    return this.props
+  }
+  context: SetupContext
+  render(): VNodeChild {}
+}
+class NoPropClassComponent extends StandardClassComponent {
+  count = ref(0)
+  changeCount(count: number) {
+    this.count.value = count
+  }
+  render() {
+    return h('div', `hello world ${this.count.value}`)
+  }
+}
+
+// @ts-expect-error changeCount expects an argument
+expectError(mount(NoPropClassComponent, {}).vm.changeCount())
+mount(NoPropClassComponent, {}).vm.changeCount(2)
+
+interface WithProp_Props {
```

Comment:
We usually don't use underscores in interface names. Maybe just `ComponentProps`?